spec/room_reservation_spec.rb

```diff
@@ -0,0 +1,209 @@
+      # According to the before block there should only be one reservation on 4/1/2019
+      expect(listed_reservations.length).must_equal 1
+      listed_reservations.each do |reservation|
+        expect(reservation).must_be_instance_of Hotel::Reservation
+      end
+    end
+  end
+end
+
+describe "Wave 2" do
+  before do
+    @hotel = Hotel::RoomReservation.new
+
+    room_id1 = 14
+    check_in1 = Date.new(2019, 4, 1)
+    check_out1 = Date.new(2019, 4, 2)
+    @reservation1 = @hotel.new_reservation(room_id1, check_in1, check_out1)
+
+    room_id2 = 15
+    check_in2 = Date.new(2019, 4, 3)
+    check_out2 = Date.new(2019, 4, 6)
+    @reservation2 = @hotel.new_reservation(room_id2, check_in2, check_out2)
+
+    room_id3 = 2
+    check_in3 = Date.new(2019, 3, 21)
+    check_out3 = Date.new(2019, 4, 3)
+    @reservation3 = @hotel.new_reservation(room_id3, check_in3, check_out3)
+  end
+
+  describe "list_available_rooms method" do
+    before do
+      @array_of_rooms = @hotel.list_available_rooms(
+        Date.new(2019, 4, 1), Date.new(2019, 4, 4)
+      )
+    end
+
+    it "returns an array of rooms" do
+      expect(@array_of_rooms).must_be_kind_of Array
+      @array_of_rooms.each do |room|
+        expect(room).must_be_kind_of Struct
+        expect(1..20).must_include room.id
+        expect(room.cost).must_equal 200
+      end
+    end
+
+    it "only returns available rooms" do
+      room2 = @hotel.rooms.find { |room| room.id == 2 }
+      room14 = @hotel.rooms.find { |room| room.id == 14 }
+      room15 = @hotel.rooms.find { |room| room.id == 15 }
+      remaining_rooms = @hotel.rooms.find_all { |room| room.id != 14 && room.id != 2 && room.id != 15 }
+
+      expect(@array_of_rooms).wont_include room2
+      expect(@array_of_rooms).wont_include room14
+      expect(@array_of_rooms).wont_include room15
+
+      remaining_rooms.each do |room|
+        expect(@array_of_rooms).must_include room
+      end
+    end
+
+    it "does not include blocked rooms" do
+      block = @hotel.new_block("fdsaj", Date.new(2019, 4, 1), Date.new(2019, 4, 6), 5, 150)
+
+      block.rooms.each do |room|
+        expect(@array_of_rooms).wont_include room
+      end
+    end
+  end
+
+  describe "new_reservation method" do
+    before do
+      @hotel = Hotel::RoomReservation.new
+
+      room_id1 = 1
+      check_in1 = Date.new(2019, 4, 1)
+      check_out1 = Date.new(2019, 4, 2)
+      @reservation1 = @hotel.new_reservation(room_id1, check_in1, check_out1)
+
+      room_id2 = 2
+      check_in2 = Date.new(2019, 4, 3)
+      check_out2 = Date.new(2019, 4, 6)
+      @reservation2 = @hotel.new_reservation(room_id2, check_in2, check_out2)
+
+      room_id3 = 3
+      check_in3 = Date.new(2019, 3, 21)
+      check_out3 = Date.new(2019, 4, 3)
+      @reservation3 = @hotel.new_reservation(room_id3, check_in3, check_out3)
+    end
+
+    it "instantiates a new Reservation and adds it to master collection" do
```

Comment:
You should also test that you can add a reservation on the same date the previous reservation ends.